Crypto/AsymmetricCrypto.cpp

```diff
@@ -0,0 +1,329 @@
+// This is an open source non-commercial project. Dear PVS-Studio, please check it.
+
+// PVS-Studio Static Code Analyzer for C, C++ and C#: http://www.viva64.com
+
+#include "pch.h"
+#include "AsymmetricCrypto.h"
+
+AsymmetricCrypto::AsymmetricCrypto()
+{
+    m_keypair = std::make_shared<EVP_PKEY*>();
+    m_rsa_encr_ctx = std::make_shared<EVP_CIPHER_CTX*>();
+    m_rsa_decr_ctx = std::make_shared<EVP_CIPHER_CTX*>();
+
+    InitializeContext();
+
+    GenerateRsaKeypair();
```

Comment:
What will happen if  GenerateRsaKeypair()  will return false? Your smart pointers will delete raw pointers to objects, but not objects themselves. Destructor will not be invoked, because constructor will not create an AsymmetricCrypto object.